WooCommerce/Classes/Tools/Currency/CurrencyFormatter.swift

```diff
@@ -124,25 +128,27 @@ public class CurrencyFormatter {
     /// Applies currency option settings to the amount (as String) for the given currency.
     ///
     /// - Parameters:
-    ///     - amount: a raw string representation of the amount, from the API, with no formatting applied. e.g. "19.87"
+    ///     - stringAmount: a raw string representation of the amount, from the API, with no formatting applied. e.g. "19.87"
     ///     - currency: a 3-letter country code for currencies that are supported in the API. e.g. "USD"
+    ///     - locale: the locale that is used to format the currency amount string.
     ///
-    func formatAmount(_ stringAmount: String, with currency: String = CurrencySettings.shared.currencyCode.rawValue) -> String? {
-        guard let decimalAmount = convertToDecimal(from: stringAmount) else {
+    func formatAmount(_ stringAmount: String, with currency: String? = nil, locale: Locale = .current) -> String? {
+        guard let decimalAmount = convertToDecimal(from: stringAmount, locale: locale) else {
             return nil
         }
 
-        return formatAmount(decimalAmount, with: currency)
+        return formatAmount(decimalAmount, with: currency ?? currencySettings.currencyCode.rawValue, locale: locale)
     }
 
 
     /// Formats the provided `amount` param into a human readable value and applies the currency option
     /// settings for the given currency.
     ///
     /// - Parameters:
-    ///   - amount: a raw string representation of the amount, from the API, with no formatting applied. e.g. "19.87"
+    ///   - stringAmount: a raw string representation of the amount, from the API, with no formatting applied. e.g. "19.87"
```

Comment:
Same here, I think we can refactor it calling it `amount`

Reply:
since we can't use the same variable created in the same `guard` statement to add `assertionFailure` as in the suggestion below, I'm going to leave this one for now (the variable is internal and I think the debugging benefit outweighs the naming here)